tasks/main.yml

```diff
@@ -7,7 +7,17 @@
   with_first_found:
     - "{{ ansible_distribution }}.yml"
     - "{{ ansible_os_family }}.yml"
-  tags: bind
+  tags: bind,pretask
+
+# JY | file to set keys for XFR authentication
+- name: create extra config file for authenticated XFR request
+  tags: pretask
+  template:
+    src: auth_transfer.j2
+    dest: /etc/bind/auth_transfer.conf
+    mode: 0640
```

Comment:
The dest: should be parameterised, based on O/S (perhaps as a parameter in vars/<O/S>.yml  since not all system's use /etc/bind as config directory.

Also, this task appears too early in the workflow, before the "master" or "slave" config has been created.

It should also only run if bind_dns_keys is defined, and not empty...